AgileVentures/sing_for_needs
# Dockerize Application  for Development and Deployment #36

#### What does this PR do?
Implements use of containers to run the application.
> Application image is based of the official elixir image (*elixir:1.8.1*)
Database image is based of official postgres image (*postgres:latest*)

#### Description of Issue to be completed?
- Create configuration to create application image
- Create configuration to run application with other required dependencies like the  database

#### How should this be manually tested?
Ensure that you have a stable version of docker on your machine

1. Change to the application directory. On terminal, run command ...
- `docker build -f docker/release/Dockerfile -t sing_for_needs:develop . `

After the application image has been created, start the application by running.
- `docker-compose -f docker/release/docker-compose.yml up `


#### Any background context you want to provide?
Docker containers ensure consistency across multiple development and release cycles, standardizing your environment. One of the biggest advantages to a Docker-based architecture is actually standardization. Docker provides repeatable development, build, test, and production environments. Standardizing service infrastructure across the entire pipeline allows every team member to work in a production parity environment. By doing this, engineers are more equipped to efficiently analyze and fix bugs within the application.

_N/B Google the benefits of using docker_

#### What are the relevant issues covered ?
#36
